File: AspNetMvcUnleashed/0672329980_source/Chapter04/VB/MvcApplication1/MvcApplication1/MyViewEngines/SimpleViewEngine.vb
Public Class SimpleViewEngine
	Inherits VirtualPathProviderViewEngine

	Public Sub New()
		Me.ViewLocationFormats = New String() { "~/Views/{1}/{0}.simple", "~/Views/Shared/{0}.simple"}
		Me.PartialViewLocationFormats = New String() { "~/Views/{1}/{0}.simple", "~/Views/Shared/{0}.simple" }
	End Sub

	Protected Overrides Function CreateView(ByVal controllerContext As ControllerContext, ByVal viewPath As String, ByVal masterPath As String) As IView
		Dim physicalPath = controllerContext.HttpContext.Server.MapPath(viewPath)
		Return New SimpleView(physicalPath)
	End Function

	Protected Overrides Function CreatePartialView(ByVal controllerContext As ControllerContext, ByVal partialPath As String) As IView
		Dim physicalPath = controllerContext.HttpContext.Server.MapPath(partialPath)
		Return New SimpleView(physicalPath)
	End Function
End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter04/CS/MvcApplication1/MvcApplication1/MyViewEngines/SimpleViewEngine.cs
using System.Web.Mvc;

namespace MvcApplication1.MyViewEngines
{
    public class SimpleViewEngine : VirtualPathProviderViewEngine
    {
        public SimpleViewEngine()
        {
            this.ViewLocationFormats = new string[] { "~/Views/{1}/{0}.simple", "~/Views/Shared/{0}.simple"};
            this.PartialViewLocationFormats = new string[] { "~/Views/{1}/{0}.simple", "~/Views/Shared/{0}.simple" };
        }

        protected override IView CreateView(ControllerContext controllerContext, string viewPath, string masterPath)
        {
            var physicalPath = controllerContext.HttpContext.Server.MapPath(viewPath);
            return new SimpleView(physicalPath);
        }

        protected override IView CreatePartialView(ControllerContext controllerContext, string partialPath)
        {
            var physicalPath = controllerContext.HttpContext.Server.MapPath(partialPath);
            return new SimpleView(physicalPath);
        }
    }
}